packages/x-data-grid/src/hooks/utils/useGridSelector.ts

```diff
@@ -127,13 +128,19 @@ export const useGridSelectorV8 = <Api extends GridApiCommon, Args, T>(
   refs.current.state = state;
   refs.current.equals = equals;
   refs.current.selector = selector;
+  const prevArgs = refs.current.args;
+  refs.current.args = args;
+  if (didInit && !refs.current.equals<typeof args>(prevArgs, args)) {
+    // React to arguments change
+    apiRef.current.store.update(apiRef.current.state);
+  }
```

Comment:
Why is this needed? I find the `store.update` surprising. This should listen to the store, not update it.

Reply:
It is to recompute the particular substate on the arguments' reference update.

We could skip this, but then the arguments change will not be reflected immediately, only on the next `setState` call which might not be the required behavior in some use cases.
As I understand, `store.update()` will only impact those subscribers whose actual state value gets modified. Ideally, the one using this particular `args` value.

Do you see any obvious drawbacks?

Reply:
`store.update()` calls every selector (scales as `O(n)`), I'd rather avoid it if there's no reason to call it.

This is just to update the current selector right? Wouldn't it be more simple to call `setState(applySelector(...etc))`?

And is `refs.current.equals` good for this use-case? Can `args` be an array?

Reply:
> This is just to update the current selector right?

I am not sure honestly, I can't think of a case where it'd affect any other selector, so yes updating the local state seems to be a better alternative.

> And is `refs.current.equals` good for this use-case? Can args be an array?

99% of the usecases should be simple ones (string, number) or objects, even for arrays the reference comparison should work fine, no?

Reply:
I think disallowing object type would reduce use case with more than one argument, and I imagine there will be multiple such use cases (for one of the use-cases, I needed to pass `rowId` and `apiRef`)

I'd rather expect the Object/Array reference to be stabilized by the user (it's mostly the internal team so it should be achievable), we could add a respective docs section when we make this API the default to explain the need to the end users too.

Wdyt?

Reply:
It would be unergonomic to stabilize the arg, it would be something like this:

```
useGridSelector(apiRef, selector, useMemo(() => [a, b, c], [a, b, c]))
```

If we're going to support objects/arrays, we should use a custom comparison function that does a shallow comparison for complex types, and `Object.is` for primitive types. Wdyt?

Reply:
It's a nice idea, I had the same in the back of my mind when using `refs.current.equals` rather than equality or `Object.is` so that users have the flexibility to pass on a custom `equals` but that would also replace it for state comparison too.

Having a custom internal function for this could be more ergonomic, yes 👍 

What would be the best way to decide which one to use, runtime decision?
```tsx
let argsEqual = Object.is;
if (!['string', 'number', 'boolean'].includes(typeof args)) {
  argsEqual = fastObjectShallowCompare;
}
if (didInit && argsEqual(prevArgs, args)) {
  // recompute state
}
```

This might not work as expected if the args type changes from primitive to complex and vice versa. How about showing a dev warning in that case?

Also, does allowing users to override the equality check make sense too? I'd avoid it at this point.

Reply:
Yes but abstract that logic in a function, `function argsEqual(a, b) { ... }`. I would ignore cases where `typeof a !== typeof b`, let the jsdoc explain that constraint. I wouldn't allow overrides.

Reply:
> let the jsdoc explain that constraint

Typescript enforces that as long as you provide the types for a specific arguments parameter. So it should be fine.

![image](https://github.com/user-attachments/assets/5107cf82-84fa-487a-bdfa-73896f1ef03a)